### PR TITLE
feat: New MyC Submit for Sale flow

### PR DESCRIPTION
**Pull request by olerichter00**
Addresses [ONYX-1071]

## Description

This implements the new My Collection "Submit for Sale" flow. 

When the new Sale flow is enabled, clicking on "Submit for Sale" creates a new submission and opens the first step in the submission flow with the submission ID. 

This PR depends on https://github.com/artsy/metaphysics/pull/5803

## Screenshots

Screenshot 2024-06-14 at 11 00 46

[ONYX-1071]: https://artsyproduct.atlassian.net/browse/ONYX-1071?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx [hunk 1]
```diff
@@ -1,149 +1,134 @@
...
   learnMore: () => void
-  slug: string
-  artworkId: string
+  artwork: MyCollectionArtworkSWASection_artwork$key
   ctaColor?: "primaryBlack" | "secondaryNeutral" | null
 }
 
-export const MyCollectionArtworkSWASectionMobileLayout: React.FC<Props> = ({
-  route,
+export const MyCollectionArtworkSWASection: React.FC<MyCollectionArtworkSWASectionProps> = ({
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:20]: No need to have separate components for desktop and mWeb. I've merged both to simplify this component.

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx [hunk 2]
```diff
@@ -1,149 +1,134 @@
...
+
+      <Button
+        variant="primaryBlack"
+        width="100%"
+        mb={2}
+        // @ts-ignore
+        as={enableNewSubmissionFlow ? undefined : RouterLink}
+        to={
+          !enableNewSubmissionFlow &&
+          `/collector-profile/my-collection/submission/artwork-details/${artwork.internalID}`
```
Comment (MounirDhahri) [inline src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:89]: we can probably remove this route as well once we release

Comment (olerichter00) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:89]: Yes, we can remove it in the end. I've already created a ticket for it: https://artsyproduct.atlassian.net/browse/ONYX-1074

Comment (MounirDhahri) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:89]: awesome

**Review by MounirDhahri — APPROVED** [review]

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx [hunk 3]
```diff
@@ -1,149 +1,134 @@
...
+    try {
+      let submissionID = artwork.consignmentSubmission?.internalID
+
+      if (!submissionID) {
+        submissionID = await createOrUpdateConsignSubmission(relayEnvironment, {
+          sessionID: !isLoggedIn ? getENV("SESSION_ID") : undefined,
+          artistID: artwork.artist?.internalID,
+          myCollectionArtworkID: artwork.internalID,
+          // Source is necessary in order to link this to a mycollection artwork
+          source: "MY_COLLECTION",
```
Comment (nickskalkin) [inline src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:59]: this can be removed now

**Review by nickskalkin — APPROVED** [review]
looks good! Left 1 small comment

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx (hunk 3 above)
Comment (olerichter00) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:59]: I suppose we do not have to set `source: "MY_COLLECTION"` anymore whenever `myCollectionArtworkID` is set when submitting an artwork from My Collection, right?

Comment (olerichter00) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:59]: Will `source` be automatically set as well when updating a submission?

Comment (olerichter00) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:59]: https://github.com/artsy/force/pull/14071

Comment (nickskalkin) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx:59]: You are right, source is set automatically only in `createConsignmentSubmission` mutation: https://github.com/artsy/metaphysics/pull/5803/files#diff-d5fd9257b7014a414628f2481fe406669003b05c3b5d0a8113cb6e67137bc7b6R129

**Pull request merged**